docs/backends/fs2.md

```diff
@@ -19,44 +28,71 @@ Next you'll need to define a backend instance as an implicit value. This can be
 
 A non-comprehensive summary of how the backend can be created is as follows:
 
-```scala
-import sttp.client.asynchttpclient.fs2.AsyncHttpClientFs2Backend
-
-AsyncHttpClientFs2Backend().flatMap { implicit backend => ... }
-
-// or, if you'd like to use custom configuration:
-AsyncHttpClientFs2Backend.usingConfig(asyncHttpClientConfig).flatMap { implicit backend => ... }
+```scala mdoc:compile-only
+AsyncHttpClientFs2Backend[IO]().flatMap { implicit backend => ??? }
+```
+or, if you'd like to use a custom configuration:
+```scala mdoc:compile-only
+import org.asynchttpclient.AsyncHttpClientConfig
 
-// or, if you'd like to use adjust the configuration sttp creates:
-AsyncHttpClientFs2Backend.usingConfigBuilder(adjustFunction, sttpOptions).flatMap { implicit backend => ... }
+val config: AsyncHttpClientConfig = ???
+AsyncHttpClientFs2Backend.usingConfig[IO](config).flatMap { implicit backend => ??? }
+```
+or, if you'd like to use adjust the configuration sttp creates:
+```scala mdoc:compile-only
+import org.asynchttpclient.DefaultAsyncHttpClientConfig
 
-// or, if you'd like the backend to be wrapped in cats-effect Resource:
-AsyncHttpClientFs2Backend.resource().use { implicit backend => ... }
+val sttpOptions: SttpBackendOptions = SttpBackendOptions.Default 
+val adjustFunction: DefaultAsyncHttpClientConfig.Builder => DefaultAsyncHttpClientConfig.Builder = ???
+AsyncHttpClientFs2Backend.usingConfigBuilder[IO](adjustFunction, sttpOptions).flatMap { implicit backend => ??? }
+```
+or, if you'd like the backend to be wrapped in cats-effect Resource:
+```scala mdoc:compile-only
+AsyncHttpClientFs2Backend.resource[IO]().use { implicit backend => ??? }
+```
+or, if you'd like to instantiate the AsyncHttpClient yourself:
+```scala mdoc:compile-only
+import org.asynchttpclient.AsyncHttpClient
 
-// or, if you'd like to instantiate the AsyncHttpClient yourself:
-implicit val sttpBackend = AsyncHttpClientFs2Backend.usingClient(asyncHttpClient)
+val asyncHttpClient: AsyncHttpClient = ??? 
+implicit val sttpBackend = AsyncHttpClientFs2Backend.usingClient[IO](asyncHttpClient)
 ```
 
 ## Using HttpClient (Java 11+)
 
 To use, add the following dependency to your project:
 
 ```
-"com.softwaremill.sttp.client" %% "httpclient-backend-fs2" % "2.2.1"
+"com.softwaremill.sttp.client" %% "httpclient-backend-fs2" % "@VERSION@"
 ```
-
-Create the backend using:
-
-```scala
+And some imports:
+```scala mdoc:reset:silent
 import sttp.client.httpclient.fs2.HttpClientFs2Backend
+import cats.effect._
+import sttp.client._
 
-HttpClientFs2Backend().flatMap { implicit backend => ... }
+// an implicit ContextShift is required to create a concurrent instance for `cats.effect.IO`,
```

Comment:
you probably mean `cats.effect.Concurrent`? this looks like an adjective, not as a type name. Same for `Blocker`